### PR TITLE
Create ImageBlockFactory

### PR DESCRIPTION
**Pull request by mgax**
Fixes https://github.com/wagtail/wagtail-factories/issues/97.

### CODE REVIEW

File: src/wagtail_factories/blocks.py [hunk 1]
```diff
@@ -239,3 +245,26 @@ def _build(cls, model_class, document):
...
+        decorative = factory.Faker("boolean")
+        alt_text = factory.Sequence(lambda n: f"Alt text {n}")
+
+        class Meta:
+            model = ImageBlock
+
+        @classmethod
+        def _construct_struct_value(cls, block_class, params):
+            if image := params.get("image"):
+                decorative = params.get("decorative")
```
Comment (jams2) [inline src/wagtail_factories/blocks.py:263]: Question: will `decorative` be in `params` if not supplied by the user? If not, should we default to `False` rather than `None`, for consistency?

File: src/wagtail_factories/blocks.py [hunk 2]
```diff
@@ -239,3 +245,26 @@ def _build(cls, model_class, document):
...
+                alt_text = params.get("alt_text")
```
Comment (jams2) [inline src/wagtail_factories/blocks.py:264]: ...and maybe the default here should be the empty string, rather than `None`? Does Wagtail insert a default of `None` or `""` for CharBlocks with `required=False`?

File: tests/test_blocks.py [hunk 3]
```diff
@@ -231,6 +232,37 @@ def test_document_chooser_block():
...
+    assert isinstance(value.contextual_alt_text, str)
+    assert value.contextual_alt_text.startswith("Alt text")
+
+
+@pytest.mark.skipif(not WAGTAIL_63_OR_HIGHER, reason="ImageBlock requires Wagtail 6.3+")
+@pytest.mark.django_db()
+def test_image_block_with_no_image():
+    value = wagtail_factories.ImageBlockFactory(image=None)
+
+    assert value is None
```
Comment (jams2) [inline tests/test_blocks.py:263]: Question: what's the rationale here?

File: src/wagtail_factories/blocks.py (hunk 1 above)
Comment (mgax) [reply src/wagtail_factories/blocks.py:263]: Yes, it will be a boolean value generated by Faker.

File: tests/test_blocks.py (hunk 3 above)
Comment (mgax) [reply tests/test_blocks.py:263]: It's because of `ImageBlockFactory._construct_struct_value`:
https://github.com/wagtail/wagtail-factories/blob/e9ead57fb8856c7beba68df33c4d7bdfaeb4d005/src/wagtail_factories/blocks.py#L262-L270

This code comes from Wagtail's [ImageBlock._struct_value_to_image](https://github.com/wagtail/wagtail/blob/v6.4/wagtail/images/blocks.py#L121C9-L130) method, and it accounts for the possibility that the image is not set (or, I imagine, it has been deleted since it was set). I chose to preserve this logic, in case the user might want to test such a scenario in their code. I've added a couple more tests to make sure that the Streamfield block gets saved correctly to the page, even if the image is `None`.

To be fair, Wagtail has to handle all edge cases, while this might be overkill for us. I'm happy to remove the `if` inside `_construct_struct_value`, or alternatively, to add an explanation to `test_image_block_with_no_image`.

File: src/wagtail_factories/blocks.py (hunk 2 above)
Comment (mgax) [reply src/wagtail_factories/blocks.py]: Same as for `decorative`: `params` will have a faked value if the user doesn't override it.
